primitives/src/policy.rs

```diff
@@ -490,19 +493,20 @@ impl Policy {
 
     /// Returns the supply at a given time (as Unix time) in Lunas (1 NIM = 100,000 Lunas). It is
     /// calculated using the following formula:
-    /// Supply (t) = Genesis_supply + Initial_supply_velocity / Supply_decay * (1 - e^(- Supply_decay * t))
-    /// Where e is the exponential function, t is the time in milliseconds since the genesis block and
-    /// Genesis_supply is the supply at the genesis of the Nimiq 2.0 chain.
+    /// Supply (t) = Genesis_supply + Initial_supply_velocity / Supply_decay * (1 - 2^(- Supply_decay * t))
+    /// Where t is the time in milliseconds since the PoW genesis block and `genesis_supply` is the supply at
+    /// the genesis of the Nimiq 2.0 chain.
     #[cfg_attr(feature = "ts-types", wasm_bindgen(js_name = supplyAt))]
-    pub fn supply_at(genesis_supply: u64, genesis_time: u64, current_time: u64) -> u64 {
-        assert!(current_time >= genesis_time);
+    pub fn supply_at(genesis_supply: u64, current_time: u64) -> u64 {
+        assert!(current_time >= Self::POW_GENESIS_TIMESTAMP);
 
-        let t = (current_time - genesis_time) as f64;
+        let t = (current_time - Self::POW_GENESIS_TIMESTAMP) as f64;
 
         let exponent = -Policy::SUPPLY_DECAY * t;
 
         let supply = genesis_supply
-            + (Self::INITIAL_SUPPLY_VELOCITY / Self::SUPPLY_DECAY * (1.0 - exp(exponent))) as u64;
+            + (Self::INITIAL_SUPPLY_VELOCITY / Self::SUPPLY_DECAY * (1.0 - pow(2.0, exponent)))
+                as u64;
```

Comment:
Looking at this again, this looks wrong.

This function returns more than the PoS `genesis_supply` at the `current_time` of the PoS genesis block.

Reply:
That is what [I commented above](https://github.com/nimiq/core-rs-albatross/pull/2682#issuecomment-2188872842):
> But if you wanted to compute the actual supply at a certain point in time, using the PoS genesis supply generates a way-too-high number, as the graph is moved up on the Y-axis by the supply difference between PoW genesis and PoS genesis.

Reply:
If you move the start of the curve to the PoW genesis, you also need to move the genesis supply there. 

_Or_, if you want to use the PoS genesis supply, you need to start the curve with the initial velocity it will have at the PoS genesis time, which is not yet fixed, so the velocity cannot be fixed yet.